project-koku/koku
# COST-3073 - fix oci compute instance usage units

## Jira Ticket

[COST-3073](https://issues.redhat.com/browse/COST-3073)

## Description

This change will
  - add conversion for usage amount and units for OCI compute instances
  - add unit filters to storage and instance-type reports

## Testing

1. Checkout Branch
2. Restart Koku and load test customer data
3. Hit endpoints
       i. http://localhost:8000/api/cost-management/v1/reports/oci/instance-types/
       ii. http://localhost:8000/api/cost-management/v1/reports/oci/instance-types/?group_by[instance_type]=*
       iii. http://localhost:8000/api/cost-management/v1/reports/oci/storage/?group_by[instance_type]=*
       
4. You should see in the respective response metadata total usage units, `Hrs` for `instance-types` endpoints and `GB-Mo` for `storage` endpoint.


## Notes

...

related failing smoke tests https://ci.ext.devshift.net/job/project-koku-koku-pr-check/6800/  are now passing

```
lib/python3.8/site-packages/iqe_cost_management/tests/rest_api/v1/test_oci_instance_reports.py::test_api_oci_instance_endpoint[report_filter] 
------------------------------------------------------------------------------------------------- live log setup -------------------------------------------------------------------------------------------------
2022-10-13 19:37:24 INFO Checking if tenant scheme exists
2022-10-13 19:37:24 INFO loaded cdapp configs from 'cdenvconfig.json'
2022-10-13 19:37:24 INFO Using <function client_obj_maker.<locals>.make_obj at 0x7f57a57a1670> object....with url http://koku-clowder-api.ephemeral-zef9un.svc:8000/api/cost-management/v1 and verify_ssl set to False
2022-10-13 19:37:24 INFO Setting auth_type to identity
2022-10-13 19:37:24 INFO Available auth types: ['basic', 'cert', 'identity', 'jwt', 'uht']
2022-10-13 19:37:24 INFO REST: http://koku-clowder-api.ephemeral-zef9un.svc:8000/api/cost-management/v1/reports/openshift/costs/
2022-10-13 19:37:24 INFO REST: METHOD=GET, request_id=None, params=None
2022-10-13 19:37:24 INFO Schema exists
------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
2022-10-13 19:37:24 INFO REST: http://koku-clowder-api.ephemeral-zef9un.svc:8000/api/cost-management/v1/reports/oci/instance-types/
2022-10-13 19:37:24 INFO REST: METHOD=GET, request_id=None, params=None
PASSED


lib/python3.8/site-packages/iqe_cost_management/tests/rest_api/v1/test_oci_instance_reports.py::test_api_oci_instance_endpoint[previous-month-yesterday] 
------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
2022-10-13 19:37:26 INFO REST: http://koku-clowder-api.ephemeral-zef9un.svc:8000/api/cost-management/v1/reports/oci/instance-types/?start_date=2022-09-01&end_date=2022-10-12
2022-10-13 19:37:26 INFO REST: METHOD=GET, request_id=None, params=None
PASSED

```
